src/netdev.c

```diff
@@ -94,52 +94,71 @@ ni_netdev_slaveinfo_destroy(ni_slaveinfo_t *slave)
 	memset(slave, 0, sizeof(*slave));
 }
 
-static void
-ni_netdev_free(ni_netdev_t *dev)
+void
+ni_netdev_reset(ni_netdev_t *dev)
 {
-	/* Clear out linkinfo */
-	ni_string_free(&dev->link.qdisc);
-	ni_string_free(&dev->link.kind);
+	if (!dev)
+		return;
+
+	/* Clear and Reset all (kernel) property data,
+	 * except of list, ref counter & marker flags */
+	ni_string_free(&dev->name);
+	dev->link.type = NI_IFTYPE_UNKNOWN;
+	dev->link.ifindex = 0;
+	dev->link.ifflags = 0;
+
+	ni_link_address_init(&dev->link.hwaddr);
+	ni_link_address_init(&dev->link.hwpeer);
 	ni_string_free(&dev->link.alias);
+
+	dev->link.mtu = 0;
+	dev->link.metric = 0;
+	dev->link.txqlen = 0;
+	dev->link.oper_state = 0;
+
 	ni_netdev_ref_destroy(&dev->link.lowerdev);
 	ni_netdev_ref_destroy(&dev->link.masterdev);
 	ni_netdev_slaveinfo_destroy(&dev->link.slave);
+
+	ni_string_free(&dev->link.qdisc);
+	ni_string_free(&dev->link.kind);
```

Comment:
Maybe it makes sense to move these in a `ni_linkinfo_reset()`?

Reply:
what about `dev->link.saved_mtu`?

Reply:
dev->link.saved_mtu is an internal thing used in wickedd only, not an exposed property.

I were thinking about `ni_linkinfo_reset()`, but it is not needed anywhere and it is IMO quite useless to reset &dev->link (= part of dev) without to reset `dev->*`.

Reply:
dev with link.ifindex == 0 is not a valid device (in the kernel or in wickedd).